### PR TITLE
Implement new `MatchWinRecord` schema

### LINKED ISSUE #298: Change `MatchWinRecord` to match new processor format
Per https://github.com/osu-tournament-rating/otr-processor/pull/45 we now have a new format for `MatchWinRecord`:

```rs
pub struct MatchWinRecord {
    pub match_id: i32,
    pub loser_roster: Vec<i32>,
    pub winner_roster: Vec<i32>,
    pub loser_points: i32,
    pub winner_points: i32,
    pub winner_team: Option<i32>,
    pub loser_team: Option<i32>,
    pub match_type: Option<MatchType>
}
```

### PR DESCRIPTION
**Pull request by hburn7**
The otr-processor uses a new `MatchWinRecord` schema as shown here https://github.com/osu-tournament-rating/otr-processor/pull/45

This is necessary to upload the stats correctly to the database.

Closes #298

### CODE REVIEW

File: API/DTOs/PlayerMatchStatsDTO.cs [hunk 1]
```diff
@@ -5,10 +5,10 @@ namespace API.DTOs;
 /// </summary>
 public class PlayerMatchStatsDTO
```
Comment (myssto) [inline API/DTOs/PlayerMatchStatsDTO.cs:6]: Lets document all of the fields here

File: API/Entities/MatchWinRecord.cs [hunk 2]
```diff
@@ -14,18 +14,17 @@ public class MatchWinRecord
     [Column("match_id")]
     public int MatchId { get; set; }
 
-    // Team arrays can represent individual players in individual matches (head to head)
-    [Column("team_blue")]
-    public int[] TeamBlue { get; set; } = [];
+    [Column("loser_roster")]
+    public int[] LoserRoster { get; set; } = [];
```
Comment (myssto) [inline API/Entities/MatchWinRecord.cs:18]: Document all of these properties as well

File: API/Entities/MatchWinRecord.cs [hunk 3]
```diff
@@ -14,18 +14,17 @@ public class MatchWinRecord
...
+    [Column("winner_points")]
+    public int WinnerPoints { get; set; }
 
-    [Column("red_points")]
-    public int RedPoints { get; set; }
+    [Column("loser_points")]
+    public int LoserPoints { get; set; }
 
     [Column("winner_team")]
     public int? WinnerTeam { get; set; }
```
Comment (myssto) [inline API/Entities/MatchWinRecord.cs:30]:
```suggestion
    public Team? WinnerTeam { get; set; }
```
Prefer usage of `Osu.Enums.Team` here and for `LoserTeam`

File: API/Migrations/OtrContextModelSnapshot.cs [hunk 4]
```diff
@@ -18,6 +18,9 @@ protected override void BuildModel(ModelBuilder modelBuilder)
 #pragma warning disable 612, 618
             modelBuilder
                 .HasAnnotation("ProductVersion", "8.0.1")
+                .HasAnnotation("Proxies:ChangeTracking", false)
+                .HasAnnotation("Proxies:CheckEquality", false)
+                .HasAnnotation("Proxies:LazyLoading", true)
```
Comment (myssto) [inline API/Migrations/OtrContextModelSnapshot.cs:23]: Is this intentional?

**Review by myssto — CHANGES_REQUESTED** [review]
Needs documentation, but everything else looks good

File: API/Migrations/OtrContextModelSnapshot.cs (hunk 4 above)
Comment (hburn7) [reply API/Migrations/OtrContextModelSnapshot.cs:23]: Not sure how this got generated..

Comment (hburn7) [reply API/Migrations/OtrContextModelSnapshot.cs:23]: The LazyLoading may have been generated by having `virtual` on Match but that isn't new.

Comment (hburn7) [reply API/Migrations/OtrContextModelSnapshot.cs:23]: Oh I think it's from this `.UseLazyLoadingProxies()`

File: API/DTOs/PlayerMatchStatsDTO.cs [hunk 5]
```diff
@@ -1,20 +1,71 @@
...
 {
-    public int Id { get; set; }
+    /// <summary>
+    /// The id of the player
+    /// </summary>
+    public int PlayerId { get; set; }
+    /// <summary>
+    /// The id of the match
+    /// </summary>
     public int MatchId { get; set; }
```
Comment (myssto) [inline API/DTOs/PlayerMatchStatsDTO.cs:15]: Add whitespace in between properties :) I wonder if there is an editorconfig setting for that

Comment (myssto) [reply API/DTOs/PlayerMatchStatsDTO.cs:15]: Also suppress any warnings about unused properties here and in the entity class

**Review by myssto — CHANGES_REQUESTED** [review]
Ok just these last couple nit-pick things. You can force merge once they get fixed

**Pull request merged**